al_aws_collector.js

```diff
@@ -179,13 +198,18 @@ class AlAwsCollector {
                 }
             },
             (asyncCallback) => {
-                this._azcollectc.register(regValues)
-                    .then(resp => {
-                        asyncCallback(null);
-                    })
-                    .catch(exception => {
-                        asyncCallback('AWSC0003 registration error: ' + exception);
-                    });
+                if (!process.env.collector_id || process.env.collector_id === 'none') {
+                    this._azcollectc.register(regValues)
+                        .then(resp => {
+                            const newCollectorId = resp.collector ? resp.collector.id : 'none';
```

Comment:
When would we use 'none' as a collector id? If the collector dosen't have an id, doesnt that indicate a registration failure which should be retried next invocation?

Reply:
Agree. We need to review this. If during checkin event we see that collector id is `none` then we need to call `register()`. But for now `collector_id` is optional and not required by CWE collector.

Reply:
`collector_id` is now can be passed via CFT parameter (default is `none`) if known, in order to reinstall a collector. Azcollect was also fixed to return collector_id for all AWS collector types